### PR TITLE
[4.0] fix wrong color of edit article

### LINKED ISSUE #23781: [4.0] Wrong color of form in edit article
### Steps to reproduce the issue
Content->Article->click on an existing article to get Articles:Edit page
Notice the color of options under 'Status' on right side.

### Expected result
The options should be colored like published->green etc. like other components.eg-
screenshot from 2019-02-05 21-19-32

### Actual result
screenshot from 2019-02-05 21-19-57

### PR DESCRIPTION
**Pull request by bahl24**
Signed-off-by: Nitish Bahl <nitishbahl24@gmail.com>

Pull Request for Issue #23781  .

### Summary of Changes
Added class `custom-select-color-state`.

### Expected result
screenshot from 2019-02-06 21-43-43

### Actual result
The values of different options is wrong due to which no color is displayed.
screenshot from 2019-02-06 22-06-34

### Documentation Changes Required
no

### CODE REVIEW

**Comment by bahl24** [comment]
I am facing difficulty in how to change the values of options, I tried adding-
`<option value="1">JPUBLISHED</option>
		<option value="0">JUNPUBLISHED</option>
		<option value="2">JARCHIVED</option>
		<option value="-2">JTRASHED</option>`
but the form stops working as administrator/components/com_content/Model/ArticleModel.php is expecting transition value.
Anyone has any idea about how to change value of options as I have already added the class which decides color?

**Comment by bahl24** [comment]
@puneet0191 @bembelimen @chmst @astridx Can you help?

**Comment by alikon** [comment]
the xml field in this case is of `type="transition"` and not `type="list"`
I would watch at https://github.com/joomla/joomla-cms/blob/4.0-dev/libraries/src/Form/Field/TransitionField.php

**Comment by dgrammatiko** [comment]
You have to adjust https://github.com/joomla/joomla-cms/blob/4.0-dev/build/media_src/system/js/fields/select-colour.es6.js
so it handles the values of the NEW transition system...

**Comment by bahl24** [comment]
@dgrammatiko In that file, the color is decided by the value only and as shown in the pic under actual result above, the current values are wrong.

**Comment by dgrammatiko** [comment]
@bahl24 exactly, that script was designed for the pre-transitions era. It needs to be refactored. Hint probably the select element needs to have a data attribute to make the distinction of plain select and the ones that used in the transitions. Should be very easy to patch...

**Comment by bahl24** [comment]
@dgrammatiko @alikon 
How can I add an attribute to the options select element. The transitionField.php returns this-
screenshot from 2019-02-08 11-35-46
On choosing value changes but on saving the form the value field is ' ', No matter what option is selected publish/archive/unpublish
screenshot from 2019-02-07 18-31-51
I added `data-type="published"` in article.xml but it is not added in select element.
So based on which condition should JS file differentiate?

File: administrator/components/com_content/forms/article.xml [hunk 1]
```diff
@@ -61,6 +61,7 @@
 			type="transition"
 			component="com_content"
 			label="COM_CONTENT_TRANSITION"
+            class="custom-select-color-state"
```
Comment (Quy) [inline administrator/components/com_content/forms/article.xml]:
```suggestion
			class="custom-select-color-state"
```

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on 4b7741dca2a7be73a6e5286173ccb9b2c687b428<br /><br />Doesn't change anything<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23782">issues.joomla.org/tracker/joomla-cms/23782</a>.</sub>

**Comment by brianteeman** [comment]
This is very out of date with the current branch.

**Comment by Quy** [comment]
Closing for now until there is a proper fix.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23782">issues.joomla.org/tracker/joomla-cms/23782</a>.</sub>

**Pull request closed without merging**